### PR TITLE
Bug 1178242 - properly preload privileged apps r=daleharvey

### PR DESCRIPTION
**Pull request by julienw**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7345cf70cfbf1c47fbbd0d223f91931b1f432130

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4a660cbccd3031584319b873aebc23873a3fd04e

File: tools/preload.py [hunk 1]
```diff
@@ -265,19 +265,22 @@ def fetch_webapp(app_url, directory=None):
...
+            appzip = ZipFile(zipappfile, 'r').read('manifest.webapp')
+            manifest = json.loads(appzip.decode('utf-8-sig'))
         else:
             logger.info('copying app...')
             shutil.copyfile(app_url, '%s%s%s' % (appname, os.sep, filename))
             metadata['manifestURL'] = ''.join(
                 [domain, path, 'manifest.webapp'])
 
-        manifest['package_path'] = ''.join(['/', filename])
-
```
Comment (julienw) [inline tools/preload.py]: this was not used anywhere AFAIK

File: tools/preload.py [hunk 2]
```diff
@@ -265,19 +265,22 @@ def fetch_webapp(app_url, directory=None):
...
             retrieve_from_url(
                 manifest['package_path'],
-                os.path.join(app_dir, filename))
+                zipappfile)
             metadata['manifestURL'] = url.geturl()
             metadata['packageEtag'] = open_from_url(path).headers['etag']
+
+            logger.info('extract manifest from zip...')
+            appzip = ZipFile(zipappfile, 'r').read('manifest.webapp')
+            manifest = json.loads(appzip.decode('utf-8-sig'))
```
Comment (julienw) [inline tools/preload.py]: we need to read the manifest from the downloaded zip file

**Comment by julienw** [comment]
it was done differently finally, please see the related bug for more information

**Pull request closed without merging**